spark-jobs/src/main/scala/filodb/downsampler/chunk/DownsamplerSettings.scala

```diff
@@ -80,13 +82,39 @@ class DownsamplerSettings(conf: Config = ConfigFactory.empty()) extends Serializ
 
   @transient lazy val exportRuleKey = downsamplerConfig.as[Seq[String]]("data-export.key-labels")
 
-  @transient lazy val exportBucket = downsamplerConfig.as[String]("data-export.bucket")
-
   @transient lazy val exportDropLabels = downsamplerConfig.as[Seq[String]]("data-export.drop-labels")
 
   @transient lazy val exportKeyToRules = {
     val keyRulesPairs = downsamplerConfig.as[Seq[Config]]("data-export.groups").map { group =>
       val key = group.as[Seq[String]]("key")
+      val tableName = group.as[String]("table")
+      val tablePath = group.as[String]("table-path")
+      val labelColumnMapping = group.as[Seq[String]]("label-column-mapping")
+        .sliding(2, 2).map(seq => (seq.head, seq.last)).toSeq
+      // Constructs dynamic exportSchema as per ExportTableConfig.
+      // final schema is a combination of columns defined in conf file plus some
+      // additional standardized columns
+      val tableSchema = {
+        // NOTE: ArrayBuffers are sometimes used instead of Seq's because otherwise
+        //   ArrayIndexOutOfBoundsExceptions occur when Spark exports a batch.
+        val fields = new mutable.ArrayBuffer[StructField](labelColumnMapping.length + 9)
```

Comment:
nit: add a comment on my we are adding `+9`. Might be better if we add this as a constant and add a description there

Reply:
Added comment in the latest commit.